### PR TITLE
[CELEBORN-1670] Avoid swallowing InterruptedException in ShuffleClientImpl

### PR DESCRIPTION
**Pull request by jiang13021**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
In the ShuffleClientImpl, methods such as pushData and pushMergedData might encounter interruptions during message transmission via the TransportClient. However, the InterruptedException may be ignored, as it is handled as a standard exception. As a result, the ShuffleClientImpl continues its operation(retry or revive) even when an InterruptedException occurs.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add UT: org.apache.celeborn.client.ShuffleClientSuiteJ#testPushDataAndInterrupted

### CODE REVIEW

**Comment by jiang13021** [comment]
Could you please review this? ping @waitinfuture @RexXiong @FMX @SteNicholas

**Review by RexXiong — APPROVED** [review]
LGTM

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -354,8 +354,12 @@ private void submitRetryPushData(
             batchId,
             newLoc,
             e);
-        pushDataRpcResponseCallback.onFailure(
-            new CelebornIOException(StatusCode.PUSH_DATA_CREATE_CONNECTION_FAIL_PRIMARY, e));
+        if (e instanceof InterruptedException) {
+          Thread.currentThread().interrupt();
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:358]: Could we maintain only the thread's interrupt state while still invoking callback.onFailure? And I've noticed some inconsistencies in how InterruptedException is handled in the Celeborn client.  In some cases it throws an exception, while in other cases it is simply ignored.

Comment (jiang13021) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:358]: Are you suggesting that we can invoke callback.onFailure(InterruptedException e) to interrupt the current thread? That seems reasonable. I believe that ignoring InterruptedException is unsafe, especially when we catch it without any specific handling.

Comment (RexXiong) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:358]: > Contributor Author

yes, better keep that behavior If there is no particular reason.

**Comment by turboFei** [comment]
reopen this PR, cc @jiang13021

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Merge to main(V0.6.0)

**Pull request closed without merging**